### PR TITLE
cmd-ref: document checkout displaying changes

### PR DESCRIPTION
**Pull request by skshetry**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please chose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

https://github.com/iterative/dvc/pull/3401 is under review. I have kept this docs to minimum, as the PR is more in relation with UI than any behavioral change. But. I see that, the behavior that it would checkout silent if there's an error was mentioned, which I have adjusted a bit. Also, some places look good to show outputs of `dvc checkout`.

### CODE REVIEW

File: public/static/docs/command-reference/checkout.md [hunk 1]
```diff
@@ -188,7 +189,7 @@ doesn't track those files; DVC does, so we must do this:
 ```dvc
 $ dvc fetch
 $ dvc checkout
-
+# TODO @skshetry: show output here
```
Comment (jorgeorpinel) [inline public/static/docs/command-reference/checkout.md]: ?

**Comment by jorgeorpinel** [comment]
OK please ping us/request reviews when ready. Thanks

**Comment by skshetry** [comment]
I tried adding outputs of `dvc checkout`, but my network has really taken a hit lately. And, I don't think, it matters much at the moment. So, I have rollbacked this to only document reference.

**Comment by shcheklein** [comment]
@skshetry could we at least remove existing wrong outputs (if any)? Or all of them silent now?

File: content/docs/command-reference/checkout.md [hunk 2]
```diff
@@ -58,9 +58,8 @@ restoring any file size will be almost instantaneous.
 > `cache.slow_link_warning` config option to `false` with `dvc config cache`.
 
 This command will fail to checkout files that are missing from the cache. In
-such a case, `dvc checkout` prints a warning message. It also lists removed
-files. Any files that can be checked out without error will be restored without
-being reported individually.
+such a case, `dvc checkout` prints a warning message. It also displays a list of
+changes made by the `checkout`.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/checkout.md]:
```suggestion
such a case, `dvc checkout` prints a warning message. It also lists the
partial progress made by the checkout.
```

File: content/docs/command-reference/checkout.md [hunk 3]
```diff
@@ -69,6 +68,8 @@ be pulled from remote storage using `dvc pull`.
 
 ## Options
 
+- `--summary` - displays summary of the changes.
```
Comment (jorgeorpinel) [inline content/docs/command-reference/checkout.md]:
```suggestion
- `--summary` - displays summary of the changes done by this command in the
  workspace.
```

**Comment by shcheklein** [comment]
@skshetry @jorgeorpinel what is the status of this, guys?

**Comment by jorgeorpinel** [comment]
@shcheklein idk but I just committed my suggestions since this has been waiting for a long time now. Should we take it over? Or we can merge it and open a separate issue for

> could we at least remove existing wrong outputs (if any)? Or all of them silent now?

**Comment by skshetry** [comment]
@jorgeorpinel, @shcheklein, sorry, this got lost in my notifications. AFAICT the docs do not have any outputs on checkouts (and, as checkouts were silent before), so, we don't need to change other parts.

**Comment by shcheklein** [comment]
@jorgeorpinel yes, let's take this over and update checkout outputs. I would not do this everywhere - only in places where `checkout` is the topic of a section/paragraph.

**Comment by jorgeorpinel** [comment]
@skshetry any chance you can move this to a branch in this repo (not fork)? I think you can edit the merge. Otherwise I may have to reopen it with a copy of your branch and somewhat lose the discussion history. Not a big deal but please in general avoid using fork branches on this repo 🙂

**Comment by skshetry** [comment]
@jorgeorpinel, the branch is not in the fork, this is in this repo. You might have got confused by `skshetry/` by namespacing. :smile:

**Comment by jorgeorpinel** [comment]
Haha 🤦‍♂ that `/` threw me off indeed. Thanks @skshetry

**Comment by jorgeorpinel** [comment]
I did merge master since this branch is kind of old, and updated the main example in the checkout cmd ref (including the actual command output).

**Comment by shcheklein** [comment]
@jorgeorpinel I think at least data versioning use case and tutorial should benefit from adding `dvc checkout` outputs (no need to run it even, no need to be precise)

File: content/docs/command-reference/checkout.md [hunk 4]
```diff
@@ -69,6 +68,9 @@ be pulled from remote storage using `dvc pull`.
 
 ## Options
 
+- `--summary` - displays summary of the changes done by this command in the
```
Comment (shcheklein) [inline content/docs/command-reference/checkout.md:71]: I would make it a bit explicit, I think even in the default mode we output a summary, so it's a shorter version ...

**Comment by shcheklein** [comment]
@jorgeorpinel are you on it?

File: content/docs/command-reference/checkout.md (hunk 4 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/checkout.md]: No summary is shown without this option. But I did improve it now.

File: content/docs/command-reference/checkout.md [hunk 5]
```diff
@@ -69,6 +68,10 @@ be pulled from remote storage using `dvc pull`.
 
 ## Options
 
+- `--summary` - in addition to checking out DVC-tracked data, display a short
```
Comment (shcheklein) [inline content/docs/command-reference/checkout.md:71]: I think it is sill confusing. This option just changes the format of the output (but even w/o it checkout outputs stats).

**Comment by shcheklein** [comment]
Also, see https://github.com/iterative/dvc.org/pull/1054#issuecomment-611806881

**Comment by shcheklein** [comment]
@jorgeorpinel @skshetry please resolve the conflicts, address comments and let's merge this already?

File: content/docs/command-reference/checkout.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/checkout.md:71]: There are no stats without this flag last time I checked. I tried it before writing this. Let me double check...

**Comment by jorgeorpinel** [comment]
Werid. I didn't get some of the notifications from this one until your last comment @shcheklein. Wrapping it up now...

File: content/docs/command-reference/checkout.md (hunk 5 above)
Comment (jorgeorpinel) [reply content/docs/command-reference/checkout.md:71]: Hm I was wrong. In fact --summary has a shorter output than regularly. Updating... ⏳

Comment (jorgeorpinel) [reply content/docs/command-reference/checkout.md]: Done in 5d29f1d7

**Comment by jorgeorpinel** [comment]
> at least data versioning use case and tutorial should benefit from adding dvc checkout outputs

Done in a8199278 (for all use cases)

Also solved merge conflicts now.

File: content/docs/command-reference/checkout.md [hunk 6]
```diff
@@ -58,18 +58,18 @@ restoring any file size will be almost instantaneous.
...
 situation. In some cases a pipeline must be reproduced (using `dvc repro`) to
-regenerate its outputs. (See also `dvc pipeline`.) In other cases the cache can
+regenerate its outputs (see also `dvc pipeline`). In other cases the cache can
 be pulled from remote storage using `dvc pull`.
 
 ## Options
 
-- `--summary` - show summary of the changes.
+- `--summary` - display a short summary of the changes done by this command in
+  the workspace, instead of a full list of change.
```
Comment (shcheklein) [inline content/docs/command-reference/checkout.md:72]: list of changes?

Comment (jorgeorpinel) [reply content/docs/command-reference/checkout.md:72]:
```suggestion
  the workspace, instead of a full list of changes.
```

**Pull request merged**